docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc

```diff
@@ -1 +1,81 @@
-= Creating secrets for your builds
\ No newline at end of file
+= Creating secrets for your builds
+
+When you build your pipeline, you might want to add tasks that require **secrets** in order to access external resources.
+
+NOTE: One such task is the link:https://github.com/redhat-appstudio/build-definitions/tree/main/task/sast-snyk-check[sast-snyk-check] task that uses the third-party service link:https://snyk.io/[snyk] to perform static application security testing (SAST) as a part of the default {ProductName} pipeline. Use this procedure to upload your snyk.io token. Name the secret `sast_snyk_task` so that the snyk task in the {ProductName} pipeline will recognize it and use it.
+
+If you want to create an application using source code from GitLab, you need to add a GitLab access token to {ProductName} __before__ you create an application. For details, see <<Creating secrets for GitLab-sourced apps>> below.
+
+.Procedure 
+
+. In {ProductName}, from the left navigation menu, select **Secrets**.
+. From the **Secrets** page, click **Add secret**.
+. Select a secret type:
+    * **Key/value secret**
+    * **Image pull secret**
+    * **Source secret**
+. For **Secret name**, enter a unique name for your secret.
+. Under **Key/value secret**, expand **Key/value 1**, then enter a key.
+. For **Upload the file with value for your key or paste its contents**, do one of the following:
+    * Click **Upload** to browse to, select, and upload the file that contains your key value.
+    * Drag the file that contains your key value into the space under **Upload**.
+    * Paste the contents of the file that contains your key value into the space under **Upload**.
+  Click **Clear** to remove the contents of the space under **Upload**.
+. Optional: Click **Add another key/value**.
+. Optional: Under **Labels**, add a label to tag or provide more context for your secret.
+. Click **Add secret**.
+
+== Creating secrets for GitLab-sourced apps
```

Comment:
@mshaposhnik Hi, can you please review a procedure that includes your content? It starts on line 28, "Creating secrets for GitLab-sourced apps". Does it all make sense for Konflux?

GitHub isn't mentioned as I was told that only apps created from a GitLab repo will require this special workflow when an access token is added _before_ an app is created. Correct me if this is wrong, please. 
Thanks!

Reply:
LGTM, thanks!